gossip/service/gossip_service.go

```diff
@@ -261,7 +261,8 @@ func (g *GossipService) DistributePrivateData(channelID string, txID string, pri
 	}
 
 	if err := handler.distributor.Distribute(txID, privData, blkHt); err != nil {
-		logger.Error("Failed to distributed private collection, txID", txID, "channel", channelID, "due to", err)
+		err := errors.Wrapf(err, "failed to distribute private collection, txID %s, channel %s", txID, channelID)
```

Comment:
you would want to use errors.WithMessage instead.  errors.Wrapf will create a fresh stack trace instead, the original err already contains the full stack.